src/components/buttons/_buttons.scss

```diff
@@ -41,13 +41,15 @@
     fill: $white;
     display: inline-flex;
     margin-right: spacing(xs);
+    margin-left: -4px;
```

Comment:
i've decided to go with negative margins, no to introduce another modifier to container class like `--with icon` and override `--small` and `--large` modifiers. still looks clean imho